src/EditorFeatures/Core/Implementation/NavigateTo/AbstractNavigateToSearchService.cs

```diff
@@ -71,23 +72,60 @@ private static string GetItemKind(DeclaredSymbolInfo declaredSymbolInfo)
             }
         }
 
-        private static MatchKind GetNavigateToMatchKind(IEnumerable<PatternMatch> matchResult)
+        private static MatchKind GetNavigateToMatchKind(bool containsDots, IEnumerable<PatternMatch> matchResult)
         {
-            // We may have matched the target string in multiple ways, but we'll answer with the
-            // "most optimistic" answer
-            if (matchResult.Any(r => r.Kind == PatternMatchKind.Exact))
+            // NOTE(cyrusn): Unfortunately, the editor owns how sorting of NavigateToItems works,
+            // and they only provide four buckets for sorting items before they sort by the name
+            // of the items.  Because of this, we only have coarse granularity for bucketing things.
+            //
+            // So the question becomes: what do we do if we have multiple match results, and we
+            // need to map to a single MatchKind.
+            //
+            // First, consider a main reason we have multiple match results.  And this happened
+            // when the user types a dotted name (like "Microsoft.CodeAnalysis.ISymbol").  Such
+            // a name would match actual entities: Microsoft.CodeAnalysis.ISymbol *and* 
+            // Microsoft.CodeAnalysis.IAliasSymbol.  The first will be an [Exact, Exact, Exact] 
+            // match, and the second will be an [Exact, Exact, CamelCase] match.  In this
+            // case our belief is that the names will go from least specific to most specific. 
+            // So, the left items may match lots of stuff, while the rightmost items will match
+            // a smaller set of items.  As such, we use the last pattern match to try to decide
```

Comment:
We don't want this because the user may do something like "M.C.ISymbol".  We don't want min here.  The left items are good for scoping. And the right item is important for matching.